### PR TITLE
Add `setuptools` to fix failing tests on Python 3.12

### PR DESCRIPTION
**Pull request by cclauss**
Like #1962 but for Python 3.12
* #1962

### CODE REVIEW

**Comment by claudep** [comment]
I guess a new pysolr release is needed to fix the issue on Python 3.12.

**Comment by cclauss** [comment]
Just needed `setuptools` on Python 3.12.

File: tox.ini [hunk 1]
```diff
@@ -28,6 +28,7 @@ deps =
     geopy==2.0.0
     coverage
     requests
+    setuptools
```
Comment (claudep) [inline tox.ini:31]: I would at least add a comment stating that the dependency could be dropped once pysolr 3.10 is released.

Comment (cclauss) [reply tox.ini:31]:
```suggestion
    setuptools; python_version >= "3.12"'  # Can be removed on pysolr >= v3.10
```

**Comment by claudep** [comment]
Thanks, what about squashing commits?

**Comment by cclauss** [comment]
I almost always `Squash and merge` unless the individual commits tell a story that is vital to preserve.

Screenshot 2024-04-30 at 11 31 59

**Review by claudep — APPROVED** [review]
Thanks!

**Pull request merged**